AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +3819,393 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingAreaAdditions)
+
+- (void)_addTrackingAreaView:(CPView)aView
+{
+    var trackingAreas = [aView trackingAreas];
+    
+    for (var i = 0; i < trackingAreas.length; i++)
+        [self _addTrackingArea:trackingAreas[i]];
+}
+
+- (void)_removeTrackingAreaView:(CPView)aView
+{
+    var trackingAreas = [aView trackingAreas];
+    
+    for (var i = 0; i < trackingAreas.length; i++)
+        [self _removeTrackingArea:trackingAreas[i]];
+}
+
+- (void)_addTrackingArea:(CPTrackingArea)trackingArea
+{
+    var trackingAreaView = [trackingArea view];
+    
+    if (![_trackingAreaViews containsObject:trackingAreaView])
+        [_trackingAreaViews addObject:trackingAreaView];
+    
+    // If CPTrackingAssumeInside option is set, put the tracking area in the _mouseEnteredStack
+    
+    if ([trackingArea options] & CPTrackingAssumeInside)
+        [_mouseEnteredStack addObject:trackingArea];
+}
+
+- (void)_removeTrackingArea:(CPTrackingArea)trackingArea
+{
+    // If mouse is in the tracking area, we remove it from the stack to avoid to fire a future mouseExited event
+    
+    [_mouseEnteredStack removeObjectIdenticalTo:trackingArea];
+    
+    var trackingAreaView = [trackingArea view];
+    
+    [_trackingAreaViews removeObjectIdenticalTo:trackingAreaView];
+}
+
+- (void)_handleTrackingAreaEvent:(CPEvent)anEvent
```

Comment:
sorry for interrupting.  but why then not to use `anArray.indexOf(object) >= 0` as default implementation for `containsObject:`?  just for consistency with Cocoa?

Best wishes,
 Ivan Tkachenko a.k.a. Ratijas.

El 19.12.2015, a las 6:26, Aparajita Fishman <notifications@github.com> escribi��:

> anArray.indexOf(object) >= 0